ClientApp/src/components/AccountSettings.jsx

```diff
@@ -0,0 +1,247 @@
+              <Input
+                type="checkbox"
+                onChange={(event) =>
+                  setPreferences({ ...preferences, vegan: event.target.checked })
+                }
+                value={preferences.vegan}
+              />
+              Vegan
+            </Label>
+          </FormGroup>
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) =>
+                  setPreferences({ ...preferences, dairy: event.target.checked })
+                }
+                value={preferences.dairy}
+              />
+              Dairy Allergy
+            </Label>
+          </FormGroup>
+
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) => setPreferences({ ...preferences, eggs: event.target.checked })}
+                value={preferences.eggs}
+              />
+              Eggs Allergy
+            </Label>
+          </FormGroup>
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) => setPreferences({ ...preferences, fish: event.target.checked })}
+                value={preferences.fish}
+              />
+              Fish Allergy
+            </Label>
+          </FormGroup>
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) =>
+                  setPreferences({ ...preferences, shellfish: event.target.checked })
+                }
+                value={preferences.shellfish}
+              />
+              Shellfish Allergy
+            </Label>
+          </FormGroup>
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) => setPreferences({ ...preferences, nuts: event.target.checked })}
+                value={preferences.nuts}
+              />
+              Nuts allergy
+            </Label>
+          </FormGroup>
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) =>
+                  setPreferences({ ...preferences, wheat: event.target.checked })
+                }
+                value={preferences.wheat}
+              />
+              Wheat allergy
+            </Label>
+          </FormGroup>
+          <FormGroup check inline>
+            <Label check>
+              <Input
+                type="checkbox"
+                onChange={(event) =>
+                  setPreferences({ ...preferences, soybean: event.target.checked })
+                }
+                value={preferences.soybean}
+              />
+              Soybean allergy
+            </Label>
+          </FormGroup>
```

Comment:
Code duplication can be reduced by extracting the creation of FormGroup component (can be the same file)

Reply:
Good point, changed in next commit

Reply:
Looks good